modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/internal/TaskManager.kt

```diff
@@ -79,7 +79,17 @@ class TaskManager @Inject constructor(val args: Args,
     }
 
     fun runTargets(passedTaskNames: List<String>, allProjects: List<Project>): RunTargetResult {
-        val taskInfos = calculateDependentTaskNames(passedTaskNames, allProjects)
+        // Check whether tasks passed at command line exist
+        passedTaskNames.forEach {
+            if (!hasTask(TaskInfo(it)))
+                throw KobaltException("Unknown task: $it")
```

Comment:
Exactly, isn't this expected behavior?

If you remove the check in `runTargets()`, then nothing is reported because `foo` is removed from `taskInfos` by `taskInfos = taskInfos.filter { hasTask(it) }`